### PR TITLE
[SPARK-35587][PYTHON][DOCS] Initial porting of Koalas documentation

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to port Koalas documentation to PySpark documentation as its initial step.
It ports almost as is except these differences:

- Renamed import from `databricks.koalas` to `pyspark.pandas`.
- Renamed `to_koalas` -> `to_pandas_on_spark`
- Renamed `(Series|DataFrame).koalas` -> `(Series|DataFrame).pandas_on_spark`
- Added a `ps_` prefix in the RST file names of Koalas documentation

Other then that,

- Excluded `python/docs/build/html` in linter
- Fixed GA dependency installataion

### Why are the changes needed?

To document pandas APIs on Spark.

### Does this PR introduce _any_ user-facing change?

Yes, it adds new documentations.

### How was this patch tested?

Manually built the docs and checked the output.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @ueshin @xinrong-databricks @itholic FYI

File: python/docs/source/development/ps_contributing.rst [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+The largest amount of work consists simply of implementing the pandas API using Spark's built-in functions, which is usually straightforward. But there are many different forms of contributions in addition to writing code:
+
+1. Use the project and provide feedback, by creating new tickets or commenting on existing relevant tickets.
+
+2. Review existing pull requests.
+
+3. Improve the project's documentation.
+
+4. Write blog posts or tutorial articles evangelizing Koalas and help new users learn Koalas.
```
Comment (itholic) [inline python/docs/source/development/ps_contributing.rst:20]: So, are we planning to rename the 'Koalas' stuffs to 'pandas API on Spark' in the separated PR ??

Comment (HyukjinKwon) [reply python/docs/source/development/ps_contributing.rst:20]: yeah, let's create JIRAs and work on that.

**Comment by HyukjinKwon** [comment]
retest this please

File: python/docs/source/development/ps_contributing.rst (hunk 1 above)
Comment (itholic) [reply python/docs/source/development/ps_contributing.rst:20]: Yes, just created: https://issues.apache.org/jira/browse/SPARK-35591

File: python/pyspark/pandas/accessors.py [hunk 2]
```diff
@@ -752,7 +752,7 @@ def transform_batch(self, func, *args, **kwargs) -> "Series":
         See Also
         --------
         DataFrame.pandas_on_spark.apply_batch : Similar but it takes pandas DataFrame as its
-        internal batch.
+            internal batch.
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/accessors.py:755]: This is a required change to inline to the item above: `DataFrame.pandas_on_spark.apply_batch`

**Comment by HyukjinKwon** [comment]
Hm, it passes in my local but fails in GA. looking into it..

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -381,6 +381,7 @@ jobs:
         # Jinja2 3.0.0+ causes error when building with Sphinx.
         #   See also https://issues.apache.org/jira/browse/SPARK-35375.
         python3.6 -m pip install 'sphinx<3.1.0' mkdocs numpy pydata_sphinx_theme ipython nbsphinx numpydoc 'jinja2<3.0.0'
+        python3.6 -m pip install sphinx_plotly_directive 'pyarrow<3.0.0' pandas plotly>=4.8
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:384]:
```suggestion
        python3.6 -m pip install sphinx_plotly_directive 'pyarrow<3.0.0' pandas 'plotly>=4.8'
```

**Comment by HyukjinKwon** [comment]
the tests passed. Should be ready for a review.

**Review by ueshin — APPROVED** [review]
LGTM.

**Comment by itholic** [comment]
LGTM.

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-databricks** [comment]
LGTM!

File: python/docs/source/getting_started/ps_videos_blogs.rst [hunk 4]
```diff
@@ -0,0 +1,130 @@
+======================
+Koalas Talks and Blogs
```
Comment (itholic) [inline python/docs/source/getting_started/ps_videos_blogs.rst:2]: Anyway, should we keep this here even if the name of talk or blog posts cannot be changed from Koalas ??

Comment (HyukjinKwon) [reply python/docs/source/getting_started/ps_videos_blogs.rst:2]: I think we should probably remove it. Let's do it separately. This PR ports the docs as is first.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**